apps/nowcasting-app/pages/index.tsx

```diff
@@ -164,29 +186,188 @@ export default function Home({ dashboardModeServer }: { dashboardModeServer: str
     isValidating: allGspSystemValidating,
     error: allGspSystemError
   } = useLoadDataFromApi<components["schemas"]["Location"][]>(`${API_PREFIX}/system/GB/gsp/`);
+
+  //////////////////////////////////////
+  // ALL GSP FORECAST DATA
+  //////////////////////////////////////
+  // Load future all GSP forecast data (every 5 minutes)
   const {
-    data: allGspForecastData,
-    isLoading: allGspForecastLoading,
-    isValidating: allGspForecastValidating,
-    error: allGspForecastError
+    data: allGspForecastFutureData,
+    isLoading: allGspForecastFutureLoading,
+    isValidating: allGspForecastFutureValidating,
+    error: allGspForecastFutureError
   } = useLoadDataFromApi<
     // TODO: see if we can fully integrate API paths/params into type safe functions
     // paths["/v0/solar/GB/gsp/forecast/all/"]["get"]["responses"]["200"]["content"]["application/json"]
     components["schemas"]["OneDatetimeManyForecastValues"][]
   >(
     // `/v0/solar/GB/gsp/forecast/all/`,
-    `${API_PREFIX}/solar/GB/gsp/forecast/all/?historic=true&compact=true`,
-    {}
+    `${API_PREFIX}/solar/GB/gsp/forecast/all/?historic=true&compact=true&start_datetime_utc=${encodeURIComponent(
+      `${forecastLastFetch30MinISO.slice(0, 19)}+00:00`
+    )}`,
+    {
+      onSuccess: (data) => {
+        const forecastHistoricStart = get30MinNow(forecastHistoricBackwardIntervalMinutes);
+        const prev30MinFromNowISO = `${get30MinNow(-30)}:00+00:00`;
+        setForecastLastFetch30MinISO(prev30MinFromNowISO);
+        setAllGspForecastHistory(
+          allGspForecastHistory
+            ? filterHistoricDataCompact(
+                [...allGspForecastHistory, ...data],
+                forecastHistoricStart,
+                prev30MinFromNowISO
+              )
+            : filterHistoricDataCompact(data, forecastHistoricStart, prev30MinFromNowISO)
+        );
+        setAllGspForecastFuture(filterFutureDataCompact(data, prev30MinFromNowISO));
+      }
+    }
   );
+  // Load historic all GSP forecast data (once, at page load)
+  const {
+    data: allGspForecastHistoricData,
+    isLoading: allGspForecastHistoricLoading,
+    isValidating: allGspForecastHistoricValidating,
+    error: allGspForecastHistoricError
+  } = useLoadDataFromApi<components["schemas"]["OneDatetimeManyForecastValues"][]>(
+    `${API_PREFIX}/solar/GB/gsp/forecast/all/?historic=true&compact=true&end_datetime_utc=${encodeURIComponent(
+      `${forecastLastFetch30MinISO.slice(0, 19)}+00:00`
+    )}`,
+    {
+      refreshInterval: 0, // Only load this once at beginning
+      onSuccess: (data) => {
+        if (!data) return;
+
+        const oldestTimestamp = getOldestTimestampFromCompactForecastValues(data);
+        const historicBackwardIntervalMinutes = getHistoricBackwardIntervalMinutes(data);
```

Comment:
I think Peter might have commented on this already, but `....BackwardIntervalMinutes` is hard to wrap my head around. Is there a way to name this function and variable that's simpler? Instead of `backwards`, maybe `reverse` could be used?

Reply:
Have renamed 👍